### PR TITLE
Add linting and run ruff

### PR DESCRIPTION
**Pull request by paddyroddy**

### CODE REVIEW

**Comment by p-j-smith** [comment]
Tests are passing!

It looks like linting if failing because there is no token in secrets for access to sourcery, although I can't check because I'm not an admin.

And the napari hub preview page is complaining about `ModuleNotFoundError: No module named 'npe2'` - does this need to be installed in the workflow?

**Comment by paddyroddy** [comment]
> Tests are passing!
> 
> It looks like linting if failing because there is no token in secrets for access to sourcery, although I can't check because I'm not an admin.
> 
> And the napari hub preview page is complaining about `ModuleNotFoundError: No module named 'npe2'` - does this need to be installed in the workflow?

Didn't mean to add `sourcery`, have removed. Will look into the `npe2` bit.

**Comment by paddyroddy** [comment]
Plugin preview working, changing the version to `@v0.1`

**Review by paddyroddy — COMMENTED** [review]
Just looked through the RP and made a small set of changes. Just seeing if tests pass.

File: examples/show_large_tree.py [hunk 1]
```diff
@@ -6,13 +6,15 @@
...
 import napari
+import numpy as np
 
 from napari_arboretum.graph import TreeNode
 
 
 def generate_binary_tree(max_depth: int):
     track_id = 0
-    nodes = [TreeNode(track_id, t=(0, 1), generation=1)]
+    t = np.array([0, 1])
```
Comment (paddyroddy) [inline examples/show_large_tree.py:16]: Nice way of doing it!

Comment (p-j-smith) [reply examples/show_large_tree.py:16]: it was actually necessary because mypy was complaining about `t` being a tuple

Comment (paddyroddy) [reply examples/show_large_tree.py:16]: And rightly so

**Pull request merged**